source/mainnet/smart-contracts/guides/migrate-contracts.rst

```diff
@@ -0,0 +1,164 @@
+.. _migrate-contracts-for-std-9:
+
+==============================================
+Migrate contracts for ``concordium-std`` 9.0.0
+==============================================
+
+With the |concordium_std|_ version ``9.0.0`` release, writing and reading smart contracts is easier than ever before.
+The new version of the standard library reduces the need for generics and traits in your init and receive methods.
+These generics and traits were there to support testing with the |test_infrastructure|_, which has been deprecated in favor of the |concordium-smart-contract-testing|_ library.
+
+This guide shows you how to migrate your contract code to use the simpler concrete types and use fewer generics.
+
+.. note::
+
+   You should only migrate your contract code if you are also ready to migrate all your tests to use the |concordium-smart-contract-testing|_ library.
```

Comment:
It's not exactly **all** your tests. Only those that invoke entrypoints directly. Can we make that distinction here please?